src/zc/lockfile/tests.py

```diff
@@ -94,6 +100,92 @@ def pid_in_lockfile():
     >>> lock.close()
     """
 
+
+def hostname_in_lockfile():
+    r"""
+    # hostname is correctly written into the lock file when it's included in the
+    # lock file content template
+    >>> import zc.lockfile
+    >>> with patch('socket.gethostname', Mock(return_value='myhostname')):
+    ...     lock = zc.lockfile.LockFile("f.lock", content_template='{hostname}')
+    >>> f = open("f.lock")
+    >>> _ = f.seek(1)
+    >>> f.read().strip() == 'myhostname'
+    True
+    >>> f.close()
+
+    Make sure that locking twice does not overwrite the old hostname:
+
+    >>> lock = zc.lockfile.LockFile("f.lock", content_template='{hostname}')
+    Traceback (most recent call last):
+      ...
+    LockError: Couldn't lock 'f.lock'
+
+    >>> f = open("f.lock")
+    >>> _ = f.seek(1)
+    >>> f.read().strip() == 'myhostname'
+    True
+    >>> f.close()
+
+    >>> lock.close()
+    """
+
+
+class TestLogger(object):
+    def __init__(self):
+        self.log_entries = []
+
+    def exception(self, msg, *args):
+        self.log_entries.append((msg,) + args)
+
+
+class LockFileLogEntryTestCase(unittest.TestCase):
+    """Tests for logging in case of lock failure"""
+    def setUp(self):
+        self.here = os.getcwd()
+        self.tmp = tempfile.mkdtemp()
+        os.chdir(self.tmp)
+
+    def tearDown(self):
+        os.chdir(self.here)
+        setupstack.rmtree(self.tmp)
+
+    def test_log_entry(self):
+        """PID and hostname are parsed and logged from lock file on failure"""
```

Comment:
I'm not a fan of docstrings on test methods, since some testrunners then print the docstring instead of the test class + method name, which makes it harder to find the test.

(I don't remember at the moment whether zope.testrunner does that.)